spec/classes/init_spec.rb

```diff
@@ -6,6 +6,7 @@
     :fqdn                   => 'monkey.example.com',
     :hostname               => 'monkey',
     :ipaddress              => '127.0.0.1',
+    :ipaddress6             => '::1',
```

Comment:
please revert these additions as the tests should still pass without them.

Reply:
Nope. It gives "error during compilation: Evaluation Error: Unknown variable: '::ipaddress6'".

As even Puppet 3.8.7 already knows the fact ipaddress6, I naively assumed assumed the ipaddress6 variable to be defined with respect to the changes I made in the init.pp code.

Of course, while I'm working on the spec tests I will as a default set ipaddress6 to '' for the default_facts, and provide a different value in the test itself.

Reply:
What if the system does not have an ipv6 address? The code should still work, which is why we shouldn't add the fact to the existing platform checks.